File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const map = new Map();
    const answer = [];
    
    for (let i = 0; i < s.length; i++) {
        if (map.has(s[i])) {
            answer.push(i - map.get(s[i]));
        } else {
            answer.push(-1);
        }
        
        map.set(s[i], i);
    }
    
    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    closest = {}
    result = []
    
    for i in range(len(s)):
        if s[i] in closest:
            result.append(i - closest[s[i]])
            closest[s[i]] = i
        else:
            closest[s[i]] = i
            result.append(-1)
            
    return result